frontend/src/views/OtherUses/AddEditOtherUses.jsx

```diff
@@ -107,6 +107,38 @@ export const AddEditOtherUses = () => {
     }
   }
 
+  const onCellValueChanged = useCallback(
+    async (params) => {
+      if (params.colDef.field === 'provisionOfTheAct') {
+        params.node.setDataValue('fuelCode', '')
+      }
+      if (
+        ['fuelType', 'fuelCode', 'provisionOfTheAct'].includes(
+          params.colDef.field
+        )
+      ) {
+        let ciOfFuel = 0
```

Comment:
Put this in a helper function so I can better understand what its trying to do

Reply:
Considering the following object:
```
{
    "fuelTypeId": 1,
    "fuelType": "Biodiesel",
    "fossilDerived": null,
    "provision1Id": null,
    "provision2Id": null,
    "defaultCarbonIntensity": 100.21,
    "fuelCodes": [
        {
            "fuelCodeId": 2,
            "fuelCode": "BCLCF124.4",
            "carbonIntensity": 3.62
        },
        {
            "fuelCodeId": 3,
            "fuelCode": "BCLCF125.4",
            "carbonIntensity": -2.14
        },
        {
            "fuelCodeId": 4,
            "fuelCode": "BCLCF138.5",
            "carbonIntensity": 4.26
        },
        {
            "fuelCodeId": 6,
            "fuelCode": "BCLCF251.2",
            "carbonIntensity": 0.35
        },
        {
            "fuelCodeId": 8,
            "fuelCode": "BCLCF274.2",
            "carbonIntensity": -15.03
        },
        {
            "fuelCodeId": 15,
            "fuelCode": "BCLCF362.1",
            "carbonIntensity": -1
        }
    ],
    "provisionOfTheAct": [
        {
            "provisionOfTheActId": 2,
            "name": "Fuel code - section 19 (b) (i)"
        },
        {
            "provisionOfTheActId": 3,
            "name": "Default carbon intensity - section 19 (b) (ii)"
        }
    ],
    "units": "L"
}
```
We are just checking if provision_of_the_act is 'Fuel code - section 19 (b) (i)', If it is the case, then we find the fuel_code associated to the fuel_type selected and provide the value to ciOfFuel. Otherwise, just take the value from defaultCarbonIntensity in fuel_type. which in this case is  100.21.